plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -2826,6 +2826,9 @@ private ManagedObjectReference getVmfsDatastore(VmwareContext context, VmwareHyp
         morDs = firstHostDatastoreSystemMO.findDatastoreByName(datastoreName);
         if (morDs == null) {
             morDs = firstHostDatastoreSystemMO.createVmfsDatastore(datastoreName, hostScsiDisk);
+        } else {
+            // in case of iSCSI/solidfire 1:1 VMFS datastore could be inaccessible
+            mountVmfsDatastore(new DatastoreMO(context, morDs), lstHosts);
```

Comment:
Workaround this checks and mounts inaccessible datastore. For others it will simply skip, for solidfire it will check and mount the datastore on the hosts.